### DISCUSSION TITLE
Preferred way to import PIXI classes

### DISCUSSION THREAD

**Discussion by IAmJulianAcosta** [pixijs/pixijs#7922] [Q&A]
Hi, this is more a question than a bug report.

I'm not sure how I should be importing PIXI in my app. I was trying to do:
 `import { FederatedPointerEvent } from 'pixi.js';`, but it didn't work, as it says that `FederatedPointerEvent` isn't exported from PIXI.

Then I found [here](https://api.pixijs.io/@pixi/events/PIXI/FederatedPointerEvent.html) that I can do:
`import { FederatedPointerEvent } from '@pixi/events';` (this works)

So I'm confused about which one should I use. Any advice would be really helpful

**Comment by andrewstart** [comment]
Some packages within the pixijs monorepo are not included in the main bundles because they are bonus features that too few would use actively. `@pixi/events` is planned for future inclusion in the main bundle, but currently is only available as a separate package. As such, you'll need to use `import { FederatedPointerEvent } from '@pixi/events';` which you already found is the working method.

**Reply by IAmJulianAcosta** [reply]
Hi! Thanks for your response!

This means that the main bundle has the most used features? Is there any advantage of using it versus using for example @pixi/app?

Also I found that there are two documentations websites: https://api.pixijs.io and https://pixijs.download/dev/docs/PIXI.html

Which one should be used?

**Reply by ivanpopelyshev** [reply]
There are two bundles. pixi.js and pixi.js-legacy. both dont have new @pixi/events. You have to use this package with one of bundles or make your own bundle. @pixi/app is just one small package, its not a bundle. If youiwant to know how app works, look at https://github.com/pixijs/pixijs/wiki/v5-Custom-Application-GameLoop . i usually write my own App class for not-hello-world projects
